docs/rfc/0004_component_groups_scale_view.md

```diff
@@ -0,0 +1,299 @@
+|                    |                                |
+| -------            |--------------------------------|
+| Feature            | Scale View of Component Groups |
+| Status             | In-progress                    |
+| Date               | 2022-09-05                     |
+| Authors            | arkbriar                       |
+| RFC PR #           |                                |
+| Implementation PR #|                                |
+|                    |                                |
+
+# **Summary**
+
+The RFC aims to provide a new CR to support the scale subresource over groups of some user-specified component of
+the RisingWave resource. With that, the users can benefit from the scale subresource and apply any tools that have such
+requirement. For example, the horizontal auto-scaling service (HPA/[KEDA](https://github.com/kedacore/keda)) will work
+after this is implemented.
+
+# **Motivation**
+
+> Why are we doing this?
+
+The [scale subresource](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#scale-subresource)
+is
+a subresource like the status but exposes three fixed fields:
+
+- `.spec.replicas` which refers to the desired replicas
+- `.status.replicas` which refers to the current replicas
+- `.status.selector` which refers to the Pod selector (but in a serialized manner)
+
+However, the current design of the `RisingWave` resource can not provide such a subresource because it is consisted of
+four independent components, and each of the component can have one or more groups. Therefore, the `kubectl scale` can
+not work directly on the `RisingWave` resource and so do other tools depend on that. And it is the main reason for
+proposing this new CR. I named it 'RisingWaveComponentScaleView'.
+
+> What use cases does it support?
+
+With the `RisingWaveComponentScaleView`, the users can set up a view (similar to the view in databases) of some groups
+of one RisingWave component, in a `RisingWave` object, to expose the scale subresource. Then the magic happens and
+everything that depends on the scale subresource shall work.
+
+Take the auto-scaler as an example, we can set up the HPA or KEDA for frontend Pods with
+a `RisingWaveComponentScaleView` that targets the frontend groups.
+
+> What is the expected outcome?
+
+1. A CR `RisingWaveComponentScaleView` and its controller, with the following features:
```

Comment:
When we create a `RisingWave` resource, does the controller will create a corresponding `RisingWaveComponentScaleView` automatically?

Reply:
Nope. It's optionally and should be created by hand.

Reply:
If not created automatically, maybe before using the `kubectl scale`, need to first create the `RisingWaveComponentScaleView`. Worry that users will miss this operation~

Reply:
In my understanding, for one `RisingWave`, we can create multiple `RisingWaveComponentScaleView`?
For example, for one `RisingWave`, we can have a `RisingWaveComponentScaleView` named "GroupAScaleView" which to scale the "group-a", another RisingWaveComponentScaleView named "GroupBScaleView" which to scale the "group-a" and a third `RisingWaveComponentScaleView` named "TotalScaleView" which we can scale two groups automatically

Reply:
Yea, it's true. But the scale view will lock the replicas fields of targeting groups, so it's not a good idea to create them automatically. We should let the users choose which groups they want to assemble to make them a unit of scaling. 

Besides, the users can perform the scaling by modifying the replicas directly upon RisingWave. The `RisingWaveScaleView` aims more at some auto controllers than the manual operations by end users.